lib/components/viewers/route-details.tsx

```diff
@@ -108,9 +108,27 @@ class RouteDetails extends Component<Props> {
             { id: 'components.RouteDetails.headsignTo' },
             { ...cur }
           )
+
+          // If there are only two total patterns, then we should rename
+          // both of them
+          if (amended.length === 1 && Object.entries(patterns).length === 2) {
+            amended[0].headsign = intl.formatMessage(
+              { id: 'components.RouteDetails.headsignTo' },
+              { ...amended[0] }
+            )
+          }
         }
 
-        amended.push(cur)
+        if (alreadyExistingIndex >= 0) {
+          // Only replace if new pattern has greater geometry
```

Comment:
I recall that if we fuse patterns, at one point there was an issue navigating from the route viewer to the pattern viewer. Specifically, the pattern viewer would not show a pattern by default. Can you check that the issue is not reintroduced?

Reply:
Are you referring to the bug that I fixed here? https://github.com/opentripplanner/otp-react-redux/pull/1137

Reply:
Yes I can confirm that this bug isn't happening anymore with this branch